client/app/components/sections/SearchPage/SearchPage.js

```diff
@@ -24,8 +25,11 @@ class SearchPage extends Component {
   }
 
   listingProps(listing, color) {
+
+    // DiscoveryAPI doesn't return UUID yet but a string
+    const listingKey = listing.id instanceof UUID ? listing.id.value : listing.id;
```

Comment:
If we use Immutable.Record to represent our UUIDs then this info about type (i.e. UUID model) needs to be available.